airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -349,56 +358,49 @@ def get_dataset_tables(self, dataset_id: str, project_id: Optional[str] = None,
         :param page_token: (Optional) page token, returned from a previous call,
             identifying the result set.
         :type page_token: str
-
-        :return: map containing the list of tables + metadata.
+        :param retry: How to retry the RPC.
+        :type retry: google.api_core.retry.Retry
+        :return: List of tables associated with the dataset.
         """
-        service = self.get_service()
-
-        optional_params = {}  # type: Dict[str, Union[str, int]]
-        if max_results:
-            optional_params['maxResults'] = max_results
-        if page_token:
-            optional_params['pageToken'] = page_token
-
-        dataset_project_id = project_id or self.project_id
+        project_id = project_id or self.project_id
 
-        return (service.tables().list(  # pylint: disable=no-member
-            projectId=dataset_project_id,
-            datasetId=dataset_id,
-            **optional_params).execute(num_retries=self.num_retries))
+        tables = Client(client_info=self.client_info).list_tables(
```

Comment:
Should we pass credentials to client?

Reply:
I am not sure about credentials but we can pass `http` as we usually do in disovery API

Reply:
We can pass credentials or http. I would prefer the credentials object, because I hope that this library will soon be rewritten to GRPC Otherwise only ADC for authorization will be used.
https://github.com/googleapis/python-cloud-core/blob/master/google/cloud/client.py#L131-L136

Reply:
Done